system_catalog/src/tables.rs

```diff
@@ -138,8 +138,8 @@ impl<M: Manager> SystemTable for Tables<M> {
             .all_catalogs()
             .map_err(|e| Box::new(e) as _)
             .context(table_engine::table::Scan { table: self.name() })?;
-        let mut builder =
-            RecordBatchWithKeyBuilder::new(self.schema.clone().to_record_schema_with_key());
+        let record_schema_key = request.projected_schema.to_record_schema_with_key();
```

Comment:
`projected_record_schema` may be better ?

Reply:
I'll try it later.

Reply:
Everything else seems OK to me.

Reply:
I think of something, you should also update the related test in `tests/cases/00_dummy` to demonstrate it .

Reply:
ah yes i forgot about it. thanks for reminding me

Reply:
hello @Rachelint , I don't quite understand what `projected_record_schema` means, I tried to modify the `Request` last night but there is no `projected_record_schema` parameter in it. Can you be a little more detailed?😥

Reply:
> hello @Rachelint , I don't quite understand what `projected_record_schema` means, I tried to modify the `Request` last night but there is no `projected_record_schema` parameter in it. Can you be a little more detailed?😥

Oh, I just think the name `record_schema_key` is  a bit strange ... and suggest to renaming it.